### PR TITLE
Add workaround for incorrect storage image formats

### PR DESCRIPTION
**Pull request by nhaehnle**
Add workaroundStorageImageFormats

It seems that there are unfortunately some applications that set storage
image formats incorrectly in SPIR-V. Such errors are exposed by the
recent commit 34c25be738d03 ("Clamp image_store dimensions.").

This workaround option will cause us to not trust the given storage image
formats.

~~Note that this is on top of #2382, which has a trivial textual conflict.~~ (That PR has been merged.)

### CODE REVIEW

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -115,6 +115,10 @@ cl::opt<Vkgc::DenormalMode> Fp32DenormalModeOpt(
                clEnumValN(Vkgc::DenormalMode::FlushToZero, "ftz", "Denormal input/output flushed to zero"),
                clEnumValN(Vkgc::DenormalMode::Preserve, "preserve", "Denormal input/output preserved")));
 
+cl::opt<bool> SpirvOverrideWorkaroundStorageImageFormats("spirv-override-workaround-storage-image-formats",
+                                                         cl::desc("Override the workaroundStorageImageFormats setting"),
```
Comment (tsymalla) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:119]: Maybe add a more descriptive text here? Something that describes the impact of the setting.

**Review by tsymalla — COMMENTED** [review]
Nit, otherwise LGTM. Thanks for working on this.

**Comment by amdvlk-admin** [comment]
### Test summary for commit c9c287e3aeff1a4d689dc3a3bcc5290ff772cc44
<details><summary>CTS tests (Failed: 0/208307)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36988/69449 (53.3%)</li>
<li>Failed: 0/69449 (0.0%)</li>
<li>Not Supported: 32461/69449 (46.7%)</li><li>Warnings: 0/69449 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37206/69417 (53.6%)</li>
<li>Failed: 0/69417 (0.0%)</li>
<li>Not Supported: 32211/69417 (46.4%)</li><li>Warnings: 0/69417 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 34225/69441 (49.3%)</li>
<li>Failed: 0/69441 (0.0%)</li>
<li>Not Supported: 35216/69441 (50.7%)</li><li>Warnings: 0/69441 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (nhaehnle) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp]: Done

**Comment by amdvlk-admin** [comment]
### Test summary for commit 78fbf27cbfe801cf7d944fe00f026b9ca787bb5b
<details><summary>CTS tests (Failed: 0/208297)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36986/69439 (53.3%)</li>
<li>Failed: 0/69439 (0.0%)</li>
<li>Not Supported: 32453/69439 (46.7%)</li><li>Warnings: 0/69439 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37206/69417 (53.6%)</li>
<li>Failed: 0/69417 (0.0%)</li>
<li>Not Supported: 32211/69417 (46.4%)</li><li>Warnings: 0/69417 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 34225/69441 (49.3%)</li>
<li>Failed: 0/69441 (0.0%)</li>
<li>Not Supported: 35216/69441 (50.7%)</li><li>Warnings: 0/69441 (0.0%)</li>
</ul></details>


</ul></details>

**Review by tsymalla — APPROVED** [review]

**Review by dstutt — APPROVED** [review]

**Pull request merged**